modal/_runtime/user_code_imports.py

```diff
@@ -76,7 +77,8 @@ def construct_webhook_callable(
 
     elif webhook_config.type == api_pb2.WEBHOOK_TYPE_WEB_SERVER:
         # Function spawns an HTTP web server listening at a port.
-        user_defined_callable()
+        server_thread = threading.Thread(target=user_defined_callable, name="web_server", daemon=True)
```

Comment:
Somewhat separate from this (but made more apparent here), but it bothers me a bit that there is no clear graceful way to shut down a web server launched this way on container shutdown. (Assigning some variable to a class and adding an exit lifecycle method would work I suppose, but that's not symmetric to the launch itself)